demos/xychart.html

```diff
@@ -61,7 +61,29 @@ <h1>XY Charts Bar with multiple category</h1>
     bar "sample bar" [52, 96, 35, 10, 87, 34, 67, 99]
     </pre>
 
-    <h1>XY Charts line with multiple category</h1>
+    <h1>XY Charts bar with multiple datasets</h1>
+    <pre class="mermaid">
+      xychart-beta
+      title "Basic xychart with multiple datasets"
+      x-axis "Relevant categories" [category1, "category 2", category3, category4]
+      y-axis Animals 0 --> 160
+      bar "dogs" [0, 60, 40, 30]
+      bar "cats" [20, 40, 50, 30]
+      bar "birds" [30, 60, 50, 30]
+    </pre>
+
+    <h1>XY Charts bar horizontal with multiple datasets</h1>
+    <pre class="mermaid">
+      xychart-beta horizontal
+      title "Basic xychart with multiple datasets"
+      x-axis "Relevant categories" [category1, "category 2", category3, category4]
+      y-axis Animals 0 --> 160
+      bar "dogs" [0, 60, 40, 30]
+      bar "cats" [20, 40, 50, 30]
+      bar "birds" [30, 60, 50, 30]
+    </pre>
+
```

Comment:
There is one issue if we remove the y-axis config then the y-axis should auto-calculate to the proper range. please check the below example.
```xychart-beta
      title "Basic xychart with multiple datasets"
      x-axis "Relevant categories" [category1, "category 2", category3, category4]
      bar "dogs" [40, 20, 40, 30]
      bar "cats" [20, 40, 50, 30]
      bar "birds" [30, 60, 50, 30]

Reply:
@subhash-halder This leads again to the topic discussed above: The axis crossing should be the start of the range - Zero in most cases or 20 in the diagram above. Can you state any reason why it is currently designed like this?

- *developer perspective*: it makes the code a lot more complicated
- *user perspective*: How this gap between the axis crossing and the range start (20 in the diagram above) has to be interpreted? If a categry has a value of 20 than the range should start at 15 or 18. It is not the task of an axis to make sure that minimum value can be displayed.

Reply:
@camueller We are using d3-axis to calculate the tick to display, and to get the scale we passed the min and max and the d3-axis decides which ticks to draw.
I think how d3-axis decides is if it can create the tick with a multiple of 5 then it uses that as it is more easily relatable to humans.

Reply:
I fixed the range calculation.